web/tests/help-page.spec.ts

```diff
@@ -0,0 +1,57 @@
+//Tests the help page
+import { test, expect, Page } from '@playwright/test';
+
+async function checkExactlyOneCurrent(page: Page) {
+    // There is only one highlighted item:
+    await expect(page.locator('span.find-highlight-current')).toHaveCount(1);
+}
+
+async function checkCurrentHighlight(page: Page, expectedPos) {
+    await checkExactlyOneCurrent(page);
+    // Check correct item:
+    const cssQuery = 'span.find-highlight.find-highlight-current.find-highlight-' + expectedPos;
+    await expect(page.locator(cssQuery)).toBeVisible();
+}
+
+test('search box in help', async ({ page }) => {
+    await page.goto('#/help');
+    const field = page.locator('#find-in-page-text');
+    const help = page.locator('#whole-help');
+    expect(await help.evaluate(e => e.scrollTop)).toBe(0);
+    await expect(field).toBeVisible();
+    // Search for the phrase "user".
+    await field.fill('user');
+    // Expect more than 5 matches.
+    await expect(page.locator('span.find-highlight').first()).toBeVisible();
+    expect(await page.locator('span.find-highlight').count()).toBeGreaterThan(5);
```

Comment:
Could we always use `await expect(...)` instead of `expect(await ...)`? Or does that not work everywhere?

Reply:
> Could we always use `await expect(...)` instead of `expect(await ...)`? Or does that not work everywhere?

Check out what I just found! This is the recommendation in the Playwright docs too! It's not just me! https://playwright.dev/docs/best-practices#use-web-first-assertions

Reply:
> Check out what I just found! This is the recommendation in the Playwright docs too! It's not just me! https://playwright.dev/docs/best-practices#use-web-first-assertions

On second thought, it's not just about where you put the `await`, but about using the smart built-in assertions. Using `toHaveCount(x)` instead of `.count()).toBe(x)`.

We could actually do that. Instead of checking that the count is >5, check that the 6th item is visible!

```suggestion
    await expect(page.locator('span.find-highlight').first()).toBeVisible();
    await expect(page.locator('span.find-highlight').nth(6)).toBeVisible(); // We have at least 6 matches.
```

Reply:
This would still be good. :point_up: